### ISSUE TITLE
Bootstrap for JVM backend

### ISSUE THREAD

**Issue by TisonKun** [perl6/nqp#494]
@MasterDuke17 @usev6 @jnthn @pmurias 

the recent build failure is because JVM backend does not recognize "strfromname" via d844d6fb65a1f4a63613f83ecc55fc3893109d71 . I think it might request bump bootstrap files for JVM backend, but have no idea to do it.

**Comment by pmurias** [comment]
You can reboostrap the JVM backend with j-bootstrap-files

**Comment by MasterDuke17** [comment]
@pmurias I just tried to do that last night, but it complains about the `strfromname` here https://github.com/perl6/nqp/blob/d844d6fb65a1f4a63613f83ecc55fc3893109d71/src/HLL/Actions.nqp#L245. I didn't get a chance to try more, but I suspect it'll involve something like manually backing out the rename (maybe just in that particular spot, and **not** committing the change), making new java bootstrap files and then committing them. Hopefully then the JVM will build on HEAD. I'm also not sure what this will mean for the truffle backend. I actually implemented `nqp::strfromname` for truffle using that name in anticipation of the rename landing on master.

**Comment by pmurias** [comment]
For truffle we don't need to do anything as it's not boostrapped yet.

**Comment by pmurias** [comment]
making a version that can handle the old and new names, making new
bootstrap files, undoing the old name support and making new bootstrap and
then committing should work

**Comment by TisonKun** [comment]
working on this, hope fix it before this night (from china)

**Comment by TisonKun** [comment]
here raise a new error, maybe there is another issue on JVM backend

I get error as below, and I try to reset to 3be0568b9044fb82b07e0f855409759f6d9a4427 which before this rename commit, and redo config to jvm and `make test` `make j-bootstrap-files` and failed. so here is another issue, which I don't know where it comes from.

```
java -cp src/vm/jvm/stage0 -Xbootclasspath/a:src/vm/jvm/stage0:nqp-runtime.jar:3rdparty/asm/asm-4.1.jar:3rdparty/asm/asm-tree-4.1.jar:3rdparty/jline/jline-1.0.jar:3rdparty/jna/jna-4.0.0.jar:src/vm/jvm/stage0/nqp.jar nqp --bootstrap --module-path=gen/jvm/stage1 --setting-path=gen/jvm/stage1 \
	    --setting=NQPCORE --no-regex-lib --target=jar \
	    --output=gen/jvm/stage1/QASTNode.jar gen/jvm/stage1/QASTNode.nqp
java.lang.RuntimeException: No such attribute '$!do' for this object
  in clone (gen/jvm/stage2/NQPCORE.setting:375)
  in specialize (gen/jvm/stage2/nqpmo.nqp:592)
  in compose (gen/jvm/stage2/nqpmo.nqp:835)
  in pkg_compose (gen/jvm/stage2/NQP.nqp:600)
  in package_def (gen/jvm/stage2/NQP.nqp:2846)
  in !reduce (gen/jvm/stage2/QRegex.nqp:1656)
  in !cursor_pass (gen/jvm/stage2/QRegex.nqp:1599)
  in package_def (gen/jvm/stage2/NQP.nqp:1242)
  in package_declarator:sym<class> (gen/jvm/stage2/NQP.nqp)
  in !protoregex (gen/jvm/stage2/QRegex.nqp:1696)
  in package_declarator (gen/jvm/stage2/NQP.nqp)
  in term:sym<package_declarator> (gen/jvm/stage2/NQP.nqp)
  in !protoregex (gen/jvm/stage2/QRegex.nqp:1696)
  in term (gen/jvm/stage2/NQPHLL.nqp)
  in termish (gen/jvm/stage2/NQPHLL.nqp)
  in EXPR (gen/jvm/stage2/NQPHLL.nqp:522)
  in statement (gen/jvm/stage2/NQP.nqp:962)
  in statementlist (gen/jvm/stage2/NQP.nqp)
  in comp_unit (gen/jvm/stage2/NQP.nqp:929)
  in TOP (gen/jvm/stage2/NQP.nqp:827)
  in parse (gen/jvm/stage2/QRegex.nqp:2303)
  in parse (gen/jvm/stage2/NQPHLL.nqp:1541)
  in execute_stage (gen/jvm/stage2/NQPHLL.nqp:1457)
  in run (gen/jvm/stage2/NQPHLL.nqp:1490)
  in <anon> (gen/jvm/stage2/NQPHLL.nqp:1493)
  in compile (gen/jvm/stage2/NQPHLL.nqp:1479)
  in eval (gen/jvm/stage2/NQPHLL.nqp:1177)
  in evalfiles (gen/jvm/stage2/NQPHLL.nqp:1434)
  in command_eval (gen/jvm/stage2/NQPHLL.nqp:1324)
  in command_line (gen/jvm/stage2/NQPHLL.nqp:1283)
  in MAIN (gen/jvm/stage2/NQP.nqp:4132)
  in <mainline> (gen/jvm/stage2/NQP.nqp:4128)
  in <anon> (gen/jvm/stage2/NQP.nqp)
make: *** [gen/jvm/stage1/QASTNode.jar] Error 1
```

**Comment by TisonKun** [comment]
to bisect the issue above,

`make test ; make j-bootstrap ; make clean ; make test` pass at b0d57e1302d261504045a4948ae12376fa4f8f44 but fail at a2f66567052e827a39cfda6d8908f62532ef3b12 .

**Comment by niner** [comment]
The "No such attribute '$!do' for this object" is fixed with commit c908d61d0191a156f5fdc97803b48244711bf035

**Comment by niner** [comment]
The strfromname rename breakage is fixed with commit 70c17426ed7c7b94ed358187c9488d15706eb107
